wp/plugin/wp.ml

```diff
@@ -18,149 +18,158 @@ module Typ = Extension.Type
 
 let name = "wp"
 
-let doc = "WP is a comparative analysis tool. It can compare two binaries \
-           and check that they behave in similar ways. It can also be used \
-           to check a single binary for certain behaviors."
+let doc = {|WP is a comparative analysis tool. It can compare two binaries
+            and check that they behave in similar ways. It can also be used
+            to check a single binary for certain behaviors.|}
 
 (* Mandatory arguments. *)
 
 let files = Cmd.arguments Typ.file
-    ~doc:"Path(s) to one or two binaries to analyze. If two binaries are \
-          specified, WP will run a comparative analysis."
+    ~doc:{|Path(s) to one or two binaries to analyze. If two binaries are
+           specified, WP will run a comparative analysis. If 0 or more than 2
+           files are give, WP will fail.|}
 
 let func = Cmd.parameter Typ.string ~aliases:["function"] "func"
-    ~doc:"Function to run the WP analysis on. If no function is specified or \
-          the function cannot be found in the binaries, the analysis will \
-          fail."
+    ~doc:{|Determines which function to verify. WP verifies a single function,
+           though calling it on the `main' function along with the `inline'
+           option will analyze the whole program. If no function is specified
+           or the function cannot be found in the binary/binaries, the analysis
+           will fail.|}
 
 (* Arguments that determine which properties CBAT should analyze. *)
 
 let precond = Cmd.parameter Typ.string "precond"
-    ~doc:"A custom precondition in SMT-LIB format to be used during the \
-          analysis. If no precondition is specified, a trivial precondition \
-          of `true' will be used."
+    ~doc:{|Allows the introduction of assertions to the beginning of a query.
+           This allows pruning of possible models. Assertions are specified in
+           the smt-lib2 format. For comparative predicates, one may refer to
+           variables in the original and modified programs by appending the
+           suffix `_orig' and `_mod' to variable names in the smt-lib
+           expression. For example, `--precond="(assert (= RDI_mod
+           #x0000000000000003)) (assert (= RDI_orig #x0000000000000003))".' If
+           no precondition is specified, a trivial precondition of `true' will
+           be used.|}
 
 let postcond = Cmd.parameter Typ.string "postcond"
-    ~doc:"A custom postcondition in SMT-LIB format to be used during the \
-          analysis. If no postcondition is specified, a trivial \
-          postcondition of `true' will be used."
+    ~doc:{|Replaces the default postcondition with the user-specified one,
```

Comment:
I'm not sure the user knows there is a default postcondition that they need to replace. Similar to "precond," I wonder if it would be simpler to say something like "Allows you to specify an assertion that WP will assume is true at the end of the function it is analyzing, using the smt-lib2 format. Similar to ..."

Reply:
You're right. The default postcondition was changed to `true`, so there really is nothing to replace. This flag will just add postconditions on top of the ones the user specifies with flags.